hydro_lang/src/graph/template.html

```diff
@@ -0,0 +1,1930 @@
+            color: #212529;
+            transform: translateY(-1px);
+        }
+        .icon-button:active {
+            transform: translateY(0);
+        }
+        .layout-select, .palette-select {
+            background: #f8f9fa;
+            border: 1px solid #dee2e6;
+            border-radius: 4px;
+            font-size: 11px;
+            padding: 4px 6px;
+            width: 80px;
+            color: #495057;
+        }
+        .layout-select:hover, .palette-select:hover {
+            border-color: #adb5bd;
+        }
+        /* Tooltip styles */
+        .tooltip {
+            position: absolute;
+            bottom: 100%;
+            left: 50%;
+            transform: translateX(-50%);
+            background: #333;
+            color: white;
+            padding: 4px 8px;
+            border-radius: 4px;
+            font-size: 10px;
+            white-space: nowrap;
+            opacity: 0;
+            pointer-events: none;
+            transition: opacity 0.2s ease;
+            margin-bottom: 4px;
+        }
+        .tooltip::after {
+            content: '';
+            position: absolute;
+            top: 100%;
+            left: 50%;
+            transform: translateX(-50%);
+            border: 4px solid transparent;
+            border-top-color: #333;
+        }
+        /* Special positioning for rightmost button tooltip to avoid cutoff */
+        .layout-controls .icon-button:last-child .tooltip {
+            left: auto;
+            right: 0;
+            transform: none;
+        }
+        .layout-controls .icon-button:last-child .tooltip::after {
+            left: auto;
+            right: 16px;
+            transform: none;
+        }
+        .icon-button:hover .tooltip {
+            opacity: 1;
+        }
+        .react-flow__node {
+            cursor: pointer;
+            border-radius: 8px;
+            box-shadow: 0 2px 4px rgba(0,0,0,0.1);
+            transition: all 0.2s ease;
+        }
+        .react-flow__node:hover {
+            transform: scale(1.02);
+            box-shadow: 0 4px 8px rgba(0,0,0,0.15);
+        }
+        /* Container node specific styles */
+        .react-flow__node.container-node {
+            display: flex;
+            align-items: center;
+            justify-content: center;
+            text-align: center;
+            font-size: 12px;
+            font-weight: 500;
+            color: #333;
+            overflow: hidden;
+            text-overflow: ellipsis;
+            white-space: nowrap;
+            padding: 4px 8px;
+            box-sizing: border-box;
+        }
+        /* Ensure container labels are readable when collapsed */
+        .react-flow__node.container-node .react-flow__node-default {
+            overflow: hidden;
+            text-overflow: ellipsis;
+            white-space: nowrap;
+            width: 100%;
+        }
+    </style>
+</head>
+<body>
+    <div id="root"></div>
+    <script type="text/babel">
```

Comment:
This is an awful lot of code to put inline, and running Babel in the browser is also not a good practice. Could we instead have the visualizer be a page on the docs website, where developers can drag-drop a file to visualize the graph? Could also have it set up to receive the graph in its URL query, so that we can directly open the visualizer.

Eventually, we'd probably want a separate `viz.hydro.run` domain, but this would be a good step in that direction.